events/git-and-github.md

```diff
@@ -0,0 +1,130 @@
+
+- Why Version Control Systems are needed?
+- Usage of basic Git commands to track their projects - Staging and Committing
+- Be able to push and pull code from Github
+- Be able to send a PR to an Open Source Repository
+
+## Before the workshop
+
+Share an instructions file like [this installation guide](https://github.com/kossiitkgp/git-and-github-workshop-2020/blob/main/installation.md), along with the poster of the workshop. The instructions should contain
+
+- How to install Git for Windows, Linux, Mac seperately
+- For Windows users, make that they install `git bash` and configure it
+- How to verify that git is installed
+- Install SSH(As Pushing via HTTPS has been deprecated)
+- Instructions on how to create a Github Account
+
+In the post mention the fact that we will be available for installation issues before 15 minutes of the workshop. Help with the installation issues before the workshop begins.
+
+## Flow of the workshop
+
+0. **How will the workshop proceed**
+
+   Explain that Git and Github are important parts of Software Development. Tell them how the workshop will proceed like when should audience ask queries. Tell the audience that doing hands-on by themselves along with the workshop will help them learn better. Encourage them to ask questions, as some of the Git concepts are non-intuitive.
+
+1. **Terminal Commands**
+
+   It might be a good idea to introduce basic linux Commands, as they will be used through out the workshop. You don't have to cover this topic in detail. Just important commands like - `ls`, `cat`, `echo` , `cd`, `mkdir`, `pwd` , `rm`would suffice. Windows users can be asked to try these commands on Git Bash. 
+
+2. **Why do we need something like Git**
+
+   Before teaching anything, it is important that audience understand why they are learning it. Start with an example, with which they can connect. Note that most of the audience might have not have collaborated on a software. You can give an example of a folder, containing so many files, which are regularly updated. Tell them that tools called "Version Control Systems" are used to track projects. Tell them the importance of maintaining History and backups. Then you can tell that how such tools are useful in Software development and conclude that Git is used to track Software projects.
+
+3. **Mental Model of Git**:
+
+   Directly jumping to the terminal would confuse the audience. It might be frustrating for audience to type commands that they don't understand. Hence, first explain how Git tracks your project. Tell that by Git stores snapshots of the project. And each snapshot is referred as commit. You can show a `git log` of a good project, to convey the idea of how snapshots are taken at every feature. 
+
+   Explain the idea of staging area very carefully and clearly. Tell them that before storing snapshots, it is important to add them to a staging area. Show them a picture depicting the workflow - 1. git add 2. git commit.
```

Comment:
Very good idea! I think it's good to show what happens in the staging area with the help of `git diff` and `git diff --cached` i.e. when a new file is created, it doesn't show up in `git diff`, when it's added to the staging area with `git add <file_name>`, the diff can be viewed with `git diff --cached`. Then if one edits an already committed file, the diff can be seen with `git diff` which doesn't show the new file. Using the staging area carefully helps a lot in development as one can put some changes in the staging area while still working on something and finally when it's ready to commit it, and thus it becomes easy to revert the newer changes.

Reply:
It's a good idea! But I am afraid that introducing `git diff` and `git diff --cached` to the first-timers, might confuse them. They have to first understand that changes are staged, and changes are committed. Hence we can restrict the usage of `git status`.

Thanks for making this point. I hadn't included `git status` in the document, though it was demonstrated in the workshop.